### PR TITLE
Remove legacy Image panel

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
Removed the Image (Legacy) panel. [See docs](https://foxglove.dev/docs/studio/panels/image) for the new Image panel.

**Description**
Deletes old code, adds permanent auto-migration
~Blocked on #6145, #6230~
Resolves FG-3216
Resolves FG-3090
Resolves FG-3405

### CODE REVIEW

File: packages/studio-base/src/services/migrateLayout/replacePanel.ts
Comment (jtbandes) [inline packages/studio-base/src/services/migrateLayout/replacePanel.ts:1]: this code is moved

File: packages/studio-base/src/services/migrateLayout/replacePanel.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  } else {
+    return {
+      ...layout,
+      first: replacePanelInLayout(layout.first, oldId, newId),
+      second: replacePanelInLayout(layout.second, oldId, newId),
+    };
+  }
+}
+
+export function replacePanel(
```
Comment (jtbandes) [inline packages/studio-base/src/services/migrateLayout/replacePanel.ts:26]: moved

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/types.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import type { Time } from "@foxglove/rostime";
+import type { Color, Point2D } from "@foxglove/studio-base/types/Messages";
+
+export type PathKey = string | number;
+
+export type CircleAnnotation = {
```
Comment (jtbandes) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/ImageMode/annotations/types.ts:10]: moved

**Comment by amacneil** [comment]
🎉

File: packages/studio-base/src/services/migrateLayout/replacePanel.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  oldId: string,
+  newId: string,
+  newConfig: Record<string, unknown>,
+): LayoutData {
+  const newPanelsState = {
+    ...panelsState,
+    configById: { ...panelsState.configById, [newId]: newConfig },
+  };
+  delete newPanelsState.configById[oldId];
+  if (newPanelsState.layout != undefined) {
```
Comment (foxymiles) [inline packages/studio-base/src/services/migrateLayout/replacePanel.ts:37]: Would immer make this a bit less convoluted?

**Review by foxymiles — APPROVED** [review]

**Review by snosenzo — APPROVED** [review]
feels good

File: packages/studio-base/src/services/migrateLayout/replacePanel.ts (hunk 3 above)
Comment (jtbandes) [reply packages/studio-base/src/services/migrateLayout/replacePanel.ts:37]: probably, but this is just being moved in this PR

**Pull request merged**